classes/announcement/AnnouncementTypeDAO.php

```diff
@@ -98,108 +80,75 @@ public function updateLocaleFields($announcementType)
      * Update an existing announcement type.
-     *
-     * @param AnnouncementType $announcementType
-     *
-     * @return bool
      */
-    public function updateObject($announcementType)
+    public function updateObject(AnnouncementType $announcementType): bool
     {
-        $returner = $this->update(
-            'UPDATE	announcement_types
-                        SET	context_id = ?
-			WHERE	type_id = ?',
-            [
-                $announcementType->getContextId(),
-                (int) $announcementType->getId()
-            ]
-        );
+        $affected = DB::table('announcement_types')
+            ->where('type_id', '=', $announcementType->getId())
+            ->update(['context_id' => $announcementType->getContextId()]);
 
         $this->updateLocaleFields($announcementType);
-        return $returner;
+        return $affected > 0;
     }
 
     /**
-     * Delete an announcement type. Note that all announcements with this type are also
-     * deleted.
-     *
-     * @param AnnouncementType $announcementType
+     * Delete an announcement type.
+     * @param bool $deleteAnnouncements Defaults to false, when true, all announcements with this type are also deleted.
      */
-    public function deleteObject($announcementType)
+    public function deleteObject(AnnouncementType $announcementType, bool $deleteAnnouncements = false): void
     {
-        return $this->deleteById($announcementType->getId());
+        $this->deleteById($announcementType->getId(), $deleteAnnouncements);
     }
 
     /**
-     * Delete an announcement type by announcement type ID. Note that all announcements with
-     * this type ID are also deleted.
-     *
-     * @param int $typeId
+     * Delete an announcement type by announcement type ID. Note that all announcements with this type ID are also deleted.
+     * @param bool $deleteAnnouncements Defaults to false, when true, all announcements with this type are also deleted.
      */
-    public function deleteById($typeId)
+    public function deleteById(int $typeId, bool $deleteAnnouncements = false): void
     {
-        $this->update('DELETE FROM announcement_type_settings WHERE type_id = ?', [(int) $typeId]);
-        $this->update('DELETE FROM announcement_types WHERE type_id = ?', [(int) $typeId]);
+        if ($deleteAnnouncements) {
+            $collector = Repo::announcement()->getCollector()->filterByTypeIds([$typeId]);
+            Repo::announcement()->deleteMany($collector);
+        }
 
-        $collector = Repo::announcement()->getCollector()->filterByTypeIds([(int) $typeId]);
-        Repo::announcement()->deleteMany($collector);
+        DB::table('announcement_types')
+            ->where('type_id', '=', $typeId)
+            ->delete();
     }
 
     /**
      * Delete announcement types by context ID.
-     *
-     * @param int $contextId
      */
-    public function deleteByContextId($contextId)
+    public function deleteByContextId(?int $contextId): void
```

Comment:
Maybe to note that if null is given as parameter, the site announcements (context_id = 0) is assumed?
Maybe we can also set the default value?
(The same for the getByContextId())

Reply:
I think it's better to not add a default value, it should be explicit.

p.s.: I'm working on a parallel PR to add the remaining foreign keys (and get rid of the `0` on the database), so updates related to it will be done there, for now I'm using `null`, another possibility that sounds interesting to my brain, but will generate more work is to accept multiple types (e.g. `int|SiteContext $contextId`) when the function is supposed to work with the site context.

Reply:
And what about having only `int` and not `?int` ?

Reply:
As the function was supporting `?int` before, I'll leave it this way to avoid extra changes.
I just can say that different parts of the codebase have a different handling for the "SITE" context, some places accept null, others 0, I'll standardize this on https://github.com/pkp/pkp-lib/issues/8333.